src/NuGet.Core/NuGet.Packaging/NupkgMetadata/NupkgMetadataFileFormat.cs

```diff
@@ -53,11 +53,11 @@ public static NupkgMetadataFile Read(TextReader reader, ILogger log, string path
             }
             catch (Exception ex)
             {
-                log.LogWarning(string.Format(CultureInfo.CurrentCulture,
+                var message = string.Format(CultureInfo.CurrentCulture,
                     Strings.Error_LoadingHashFile,
-                    path, ex.Message));
+                    path, ex.Message);
 
-                throw;
+                throw new Exception(message);
```

Comment:
It's also good practice to throw specific exceptions, not use the base `Exception` type.  Perhaps an `ArgumentException` can be used. From a quick look at the [SystemException docs' list of derived types](https://learn.microsoft.com/en-us/dotnet/api/system.systemexception?view=net-8.0), I don't see anything else that looks better, but I only had a very quick look.